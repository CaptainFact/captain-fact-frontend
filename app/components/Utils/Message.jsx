import React from 'react'


const Message = ({type='info', header='', children}) =>
  <article className={`message is-${type}`}>
    {header && <div className="message-header">
      {header}
    </div>}
    <div className="message-body">
      {children}
    </div>
  </article>

export default Message