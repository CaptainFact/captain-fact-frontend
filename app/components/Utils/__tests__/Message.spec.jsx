import Message from '../Message'

test("render message", () => {
  snapshot(<Message>Hello</Message>)
})

test("can set type", () => {
  snapshot(<Message type="warning">Alert !</Message>)
})

test("can set header", () => {
  snapshot(<Message header="Awesome title">Hellooow</Message>)
})

test("without body", () => {
  snapshot(<Message/>)
})
