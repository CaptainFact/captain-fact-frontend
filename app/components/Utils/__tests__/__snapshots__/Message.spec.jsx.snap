// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can set header 1`] = `
<article
  className="message is-info"
>
  <div
    className="message-header"
  >
    Awesome title
  </div>
  <div
    className="message-body"
  >
    Hellooow
  </div>
</article>
`;

exports[`can set type 1`] = `
<article
  className="message is-warning"
>
  <div
    className="message-body"
  >
    Alert !
  </div>
</article>
`;

exports[`render message 1`] = `
<article
  className="message is-info"
>
  <div
    className="message-body"
  >
    Hello
  </div>
</article>
`;

exports[`without body 1`] = `
<article
  className="message is-info"
>
  <div
    className="message-body"
  />
</article>
`;
